ArduCopter/Copter.cpp

```diff
@@ -381,6 +381,18 @@ bool Copter::set_target_angle_and_climbrate(float roll_deg, float pitch_deg, flo
     mode_guided.set_angle(q, Vector3f{}, climb_rate_ms*100, false);
     return true;
 }
+
+bool Copter::set_thrust(float thrust)
+{
+    // exit if vehicle is not in Guided mode or Auto-Guided mode
+    if (!flightmode->in_guided_mode()) {
+        return false;
+    }
+
+    Quaternion q {};
+    mode_guided.set_angle(q, Vector3f{}, thrust, true);
```

Comment:
I would prefer to add a binding for this full method so it can be used for other things as well.

Reply:
the climb_rate_cms_or_thrust is pretty horrible - might be worth trying to come up with a better API

Reply:
We can pass const argument in bindings, so we could bind with the true hardcoded and rename to something more useful without having to change copter.

Reply:
we could, but I also want bindings that make sense for quadplane. I think we need to think about what guided bindings we want and design a set that are really usable, not just match the current C++ ones for copter, as they are just so weird

Reply:
related to this is the GUID_TIMEOUT hackery, we need a way to set the timeout from scripts properly

Reply:
In general a set_target_attitude_and_thrust() would be more generally useful.